core/store/migrate/migrations/0199_add_finalized_block_number.sql

```diff
@@ -0,0 +1,10 @@
+-- +goose Up
+ALTER TABLE evm.log_poller_blocks
+    ADD COLUMN last_finalized_block_number
+        bigint not null
+        default 0
```

Comment:
Actually I think we can provide a set of update scripts like (per chain, using finalityDepth we have)
```sql
update last_finalized_block set blockNumber - chain.finalityDepth 
where last_finalized_block = 0 and chainId = X
```

Reply:
It should address this edge case
https://github.com/smartcontractkit/chainlink/pull/10762/files#r1345523740

Reply:
+1 to backfilling the data and avoiding thinking about such edge cases. Note that we may want to sanity check the perf of that with 100k blocks (default keepBlocks) but I suspect it'd be fine

Reply:
When it comes to backfilling I think we have two options here

Use raw SQL and copy and paste finalityDepth values for each chain from the `toml` configs
```sql
WITH variables AS (
    SELECT
        evm_chain_id,
        CASE
            WHEN evm_chain_id = 1 THEN 50
            WHEN evm_chain_id = 5 THEN 50
        -- Add more conditions for other evm_chain_id values
            ELSE 0
            END AS finality_depth
    FROM evm.log_poller_blocks
    GROUP BY evm_chain_id
)

UPDATE evm.log_poller_blocks AS lpb
SET last_finalized_block_number = lpb.block_number - v.finality_depth
FROM variables v
WHERE lpb.evm_chain_id = v.evm_chain_id
  AND lpb.last_finalized_block_number = 0;
```
It's very simple but it relies on hardcoded values. Meaning that if NOP has other finalityDepth settings than the config (because they can be overridden) we will use wrong finalityDepth, not sure if this is a big deal and if products actually have custom values defined or indeed rely on default config

2. Use Go migration and fetch these values from config directly during migration
```go
const backfillQuery = `
	UPDATE evm.log_poller_blocks 
	SET last_finalized_block_number = greatest(block_number - $1, 0)
	WHERE evm_chain_id = $2
	  AND last_finalized_block_number = 0;
`

func init() {
	goose.AddMigrationContext(Up200, Down200)
}

// nolint
func Up200(ctx context.Context, tx *sql.Tx) error {
	chainsWithFinality, set := os.LookupEnv(env.EVMChainFinalityDepthMigration0200)
	if !set {
		return nil
	}

	for _, chainWithFinality := range strings.Split(chainsWithFinality, ",") {
		chainID, finality, err := parseChainWithFinality(chainWithFinality)
		if err != nil {
			return err
		}

		_, err = tx.Exec(backfillQuery, finality, chainID)
		if err != nil {
			return err
		}
	}
	return nil
}
```

More error-prone, because of passing finalityDepths via env variable. Failing migration will prevent the node from starting, but we will get exact values here. Any opinions?